Mathlib/Topology/Algebra/InfiniteSum/Nonarchimedean.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2024 Mitchell Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mitchell Lee
+-/
+import Mathlib.Topology.Algebra.InfiniteSum.Group
+import Mathlib.Topology.Algebra.Nonarchimedean.Basic
+
+/-!
+# Infinite sums and products in nonarchimedean abelian groups
+
+Let `G` be a complete nonarchimedean abelian group and let `f : α → G` be a function. We prove that
+`f` is unconditionally summable if and only if `f a` tends to zero on the cofinite filter on `α`.
+We also prove the analogous result in the multiplicative setting.
+
+-/
+
+open Filter Topology BigOperators
+
+namespace NonarchimedeanGroup
+
+/--Let `G` be a nonarchimedean multiplicative abelian group, and let `f : α → G` be a function that
+tends to one on the filter of cofinite sets. For each finite subset of `α`, consider the partial
+product of `f` on that subset. These partial products form a Cauchy filter. -/
+@[to_additive "Let `G` be a nonarchimedean additive abelian group, and let `f : α → G` be a function
+that tends to zero on the filter of cofinite sets. For each finite subset of `α`, consider the
+partial sum of `f` on that subset. These partial sums form a Cauchy filter."]
+theorem cauchy_partial_prod_of_tendsto_cofinite_one {α G : Type*} [CommGroup G] [UniformSpace G]
```

Comment:
```suggestion
theorem cauchySeq_prod_of_tendsto_cofinite_one {α G : Type*} [CommGroup G] [UniformSpace G]
```
I like the new formulation using `CauchySeq`, please also update the name (and you can remove `_partial` from the name).